src/ripple/consensus/Validations.h

```diff
@@ -719,21 +727,35 @@ class Validations
         std::lock_guard lock{mutex_};
         if (toKeep_)
         {
-            for (auto i = byLedger_.begin(); i != byLedger_.end(); ++i)
+            // We only need to refresh the keep range when it's just about to
+            // expire. Track the next time we need to refresh.
+            static std::chrono::steady_clock::time_point refreshTime;
+            if (auto const now = byLedger_.clock().now(); refreshTime <= now)
             {
-                auto const& validationMap = i->second;
-                if (!validationMap.empty() &&
-                    validationMap.begin()->second.seq() >= toKeep_)
+                // The next refresh time is shortly before the expiration
+                // time from now.
+                refreshTime = now + parms_.validationSET_EXPIRES -
+                    parms_.validationFRESHNESS;
+
+                for (auto i = byLedger_.begin(); i != byLedger_.end(); ++i)
                 {
-                    byLedger_.touch(i);
+                    auto const& validationMap = i->second;
+                    if (!validationMap.empty())
+                    {
+                        auto const seq = validationMap.begin()->second.seq();
+                        if (seq >= toKeep_->low_ && seq < toKeep_->high_)
```

Comment:
Style nit: When there are multiple comparisons like this, I like to have the comparison operators all "pointing" in the same direction, and in cases like this where you're checking a range, I like to have the range boundaries on the "outside". i.e.
```
if (toKeep_->low_ <= seq && seq < toKeep_->high_)
```
I find that more readable, and easier to reason about what's being checked.

Reply:
Thanks, changed. It is more readable. I spent less time staring at the line after the change.